### PR TITLE
Reworks how a lot of glassblowing, ceramic working, and forging work - Makes them make a lot of the same stuff with different material datums - Forging weapon balance is a fuck - Ashwalkers can finally make oven trays

### PR DESCRIPTION
**Pull request by Paxilmaniac**
## About The Pull Request

This pr does a LOT, but I'll go through it all one thing at a time.

**Forging changes:**

This is where the most changes have been made, the process of forging has been simplified a good bit in many areas.

Forges no longer have differences between fuels, just add fuel and burns for an extra x amount of time.

Reagent imbuing is GONE, no more.

We do, however, have the return of material datums having effects on the stats of weapons and armor made from it.

Damages and armor on basically all of the equipment has been tweaked or redone entirely, though the two most notable changes are:
-Spears have good throwing damage again, but no longer have 2 tile reach
-Armor can no longer have goliath plates put on it

The primitive centrifuge has been removed because we recently got the primitive chem separator from tg, these have been placed in both ashwalker dens.

Also all the structures got new sprites AGAIN :)

The actual selection of things you can make with forging has been changed, there are less weapons, some of the more useless weapons are gone, and a selection of other items have been added (like dinnerware, finally I can eat off of a silver platter).

**Ceramics changes:**

The exact selection of items has been redone as well, rather than just:

Bowl, plate, cup

You instead have:

Bowl, cup, all three plate types, oven trays

**Glassblowing changes:**

Basically the same as the ceramics changes, new selection of items the same as above, but with a bottle as well!

But, for all my maintainer friends out there, I made the recipe selection system less awful :)

## How This Contributes To The Skyrat Roleplay Experience

On the forging changes:

Reagent forging was just bad, it was bad the moment we had to make a blacklist for what reagents you can and can't put in it. Being able to just apply chems on hit for basically free no matter what is not something and not something that should stay, and thus it won't be.

This leaves forging pretty bare afterwards, so with some tweaks in order to avoid how the last test of this went, material datums on forged items are once again allowed to affect the stats of weapons and armor.

But, I was also not a fan of how forging was really just nothing but weapons and armor at this point, you can make so much more stuff with metal so I put a few of those things in, various dinnerware pieces. Finally have the ability to make dinner trays, impress your friends with an all gold buffet set, so on.

The weapon and armor rebalance was done in part because of the return of material datums changing stats, and also in part because forging weapon balance is a little goofy ahh. Two tile spears, the ability to put FOUR goliath plates on the clothing, shit like that. That all has gone, and the weapon selection has been trimmed down and refined to go with it.

On glassblowing and ceramics:

Both of the system were and still kinda are complex for what they currently are, and that is making a grand selection of like 3-5 items each. I've just taken the liberty to expand the selection a bit, and de-shit the code related to these two systems a bit at the same time.

## Proof of Testing

<details>
<summary>Screenshots/Videos</summary>

image
  
</details>

## Changelog

:cl:
del: The ability to imbue reagents into forged items has been removed due to being terrible.
qol: To the relief of ashwalkers everywhere, there are now several primitive ways to make various dinnerware items, such as oven trays, plates both large and small, and so on.
balance: Forged weapons and armor have had a good bit of a rework, some items of note are material datums changing their stats again, the removal of putting goliath plates on the armor, and various weapon tweaks like removing two tile reach from the spears and giving some weapons an actual purpose.
code: The way that the glassblowing and ceramics systems choose their recipes and results now has been made much easier to add and remove to/from, begone terrible list of switch cases and welcome recipe datums!
/:cl:

### CODE REVIEW

File: modular_skyrat/modules/primitive_production/code/forging/forge_weapons.dm [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+	worn_icon_state = "dagger_back"
+	belt_icon_state = "dagger_belt"
+	hitsound = 'sound/weapons/bladeslice.ogg'
+	throwforce = 10
+	demolition_mod = 0.5
+	slot_flags = ITEM_SLOT_BELT | ITEM_SLOT_BACK
+	w_class = WEIGHT_CLASS_SMALL
+	attack_verb_continuous = list("attacks", "slashes", "stabs", "slices", "tears", "lacerates", "rips", "dices", "cuts")
+	attack_verb_simple = list("attack", "slash", "stab", "slice", "tear", "lacerate", "rip", "dice", "cut")
+	sharpness = SHARP_EDGED
```
Comment (YakumoChen) [inline modular_skyrat/modules/primitive_production/code/forging/forge_weapons.dm:56]: needs to have TOOL_KNIFE added

File: modular_skyrat/modules/primitive_production/readme.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+### Master file additions
+
+-   N/A
+
+### Included files that are not contained in this module:
+
+-   N/A
+
+### Credits:
```
Comment (jjpark-kb) [inline modular_skyrat/modules/primitive_production/readme.md:26]: so no credit to original author?

**Review by jjpark-kb — CHANGES_REQUESTED** [review]
material datums should not return, and reagent imbuing should not be removed.

File: modular_skyrat/modules/primitive_production/readme.md (hunk 2 above)
Comment (Paxilmaniac) [reply modular_skyrat/modules/primitive_production/readme.md:26]: Good catch actually

**Review by jjpark-kb — CHANGES_REQUESTED** [review]
I'm not going to accept this PR until it is atomized as well.

**Comment by Nerev4r** [comment]
> material datums should not return, and reagent imbuing should not be removed.

Would you like to explain these things at all? It would help to know why exactly this is being blocked, rather than extremely vague commands.


That being said, material datums returning is gonna be kinda weird because I know already I'm going to be seeing people strutting around with uranium armor and titanium weapons purely because they confer the best stats, and nothing else.

**Comment by Paxilmaniac** [comment]
I can atomize this on that jake is correct

**Pull request closed without merging**